docs/SdkVersioning.md

```diff
@@ -0,0 +1,590 @@
+
+
+### Release Candidates
+
+It is possible to promote a Development Release to a Release Candidate. This should be used 
+in case:
+- Stabilization is needed before you want to release this version.
+- Development on the `main` branch continued with changes not desired in the release
+
+Each Release Candidate is represented by:
+- a GitHub Release (pre-release state), and
+- an associated git tag on the specified commit.
+
+A Release Candidate is initiated on the `main` development branch;
+
+| Current Version | Increment Type      | Next Version |
+| --------------- | ------------------- | ------------ |
+| `0.1.0-dev1`    | Release Candidate   | `0.1.0-rc1`  |
+| `0.1.0`         | Release Candidate   | `0.2.0-rc1`  |
+
+Stabilizing a Release Candidate is handled on a release branch (i.e. `release/0.1.0`);
+
+| Current Version | Increment Type      | Next Version |
+| --------------- | ------------------- | ------------ |
+| `0.1.0-rc1`     | Non-breaking change | `0.1.0-rc2`  |
+| `0.1.0-rc1`     | Breaking change     | N/A          |
+
+### Releases
+
+Each Release is represented by:
+- a GitHub Release (pre-release state), and
+- an associated git tag on the specified commit.
+
+You can release both Development Releases and Release Candidates from either the `main`- or `release`- branch;
+
+| Current Version | Increment Type      | Next Version |
+| --------------- | ------------------- | ------------ |
+| `0.1.0-dev1`    | Release             | `0.1.0`      |
+| `0.1.0-rc1`     | Release             | `0.1.0`      |
+| `0.1.0`         | Release             | N/A          |
+
+Additionally, you can introduce additional fixes on a release by adding commits on a release branch (i.e. `release/0.1.0`);
+
+| Current Version | Increment Type      | Next Version |
+| --------------- | ------------------- | ------------ |
+| `0.1.0`         | Non-breaking change | `0.1.1`      |
+| `0.1.0`         | Breaking change     | N/A          |
+
+
+## Branching
+
+Below you can find an example of common strategies which can be utilized using SdkVer:
+
+```mermaid
+gitGraph
+  commit id: "0.1.0-dev1"
+  commit id: "0.1.0-dev2" tag: "0.1.0"
+  commit id: "0.2.0-dev1"
+  commit id: "0.2.0-dev2" tag: "0.2.0-rc1"
+  branch release/v0.2.0
+  checkout main
+  commit id: "0.3.0-dev1"
+  checkout release/v0.2.0
+  cherry-pick id: "0.3.0-dev1" tag: "0.2.0-rc2 + 0.2.0"
+  checkout main
+  commit id: "0.3.0-dev2"
+  commit id: "0.3.0-dev3"
+  checkout release/v0.2.0
+  cherry-pick id: "0.3.0-dev3" tag: "0.2.1"
+  checkout main
+  commit id: "1.0.0-dev1" type: REVERSE
+  commit id: "1.0.0-dev2" tag: "1.0.0-rc1"
+  branch release/v1.0.0
+  checkout main
+  commit id: "1.1.0-dev1"
+  commit id: "1.1.0-dev2"
+  checkout release/v1.0.0
+  cherry-pick id: "1.1.0-dev2" tag: "1.0.0-rc2"
+  checkout main
+  commit id: "1.1.0-dev3"
+  commit id: "1.1.0-dev4"
+  checkout release/v1.0.0
+  cherry-pick id: "1.1.0-dev4" tag: "1.0.0-rc3 + 1.0.0"
```

Comment:
I am not sure if it is clear in picture now that single commit is having two tags (both `1.0.0-rc3` and `1.0.0`).

Reply:
Unfortunately I cannot add multiple tag icons using `mermaid`, therefore opted for this solution